src/hotspot/cpu/aarch64/templateTable_aarch64.cpp

```diff
@@ -3568,12 +3568,25 @@ void TemplateTable::_new() {
 
     // The object is initialized before the header.  If the object size is
     // zero, go directly to the header initialization.
-    __ sub(r3, r3, sizeof(oopDesc));
+    if (UseCompactObjectHeaders) {
+      __ sub(r3, r3, oopDesc::base_offset_in_bytes());
+    } else {
+      __ sub(r3, r3, sizeof(oopDesc));
+    }
     __ cbz(r3, initialize_header);
 
     // Initialize object fields
     {
-      __ add(r2, r0, sizeof(oopDesc));
+      if (UseCompactObjectHeaders) {
+        __ add(r2, r0, oopDesc::base_offset_in_bytes());
+        if (!is_aligned(oopDesc::base_offset_in_bytes(), BytesPerLong)) {
```

Comment:
I see `oopDesc::base_offset_in_bytes()` must be 8 bytes, so codes here `if(){}` may be useless, right?